lnwallet/channel.go

```diff
@@ -292,6 +292,41 @@ type commitment struct {
 	incomingHTLCIndex map[int32]*PaymentDescriptor
 }
 
+// commitKeySet holds all derived keys needed to construct commitment and HTLC
+// transactions. The keys are derived differently depending whether the
+// commitment transaction is ours or the remote peer's. Private keys associated
+// with each key may belong to the commitment owner or the "other party" which
+// is referred to in the field comments, regardless of which is local and which
+// is remote.
+type commitKeySet struct {
+	// point is the "per commitment point" used to derive the tweak for each
+	// base point.
+	point *btcec.PublicKey
+
+	// localKeyTweak is the tweak used to derive the local public key from the
+	// local payment base point or the local private key from the base point
+	// secret. This may be included in a SignDescriptor to generate signatures
+	// for the local payment key.
+	localKeyTweak []byte
+
+	// delayKey is the commitment transaction owner's key which is included in
+	// HTLC success and timeout transaction scripts.
+	delayKey *btcec.PublicKey
+
+	// paymentKey is the other party's payment key in the commitment tx.
+	paymentKey *btcec.PublicKey
```

Comment:
Seems redundant as `remoteKey` is defined below?

Reply:
`paymentKey` is assigned to `localKey` if generating the remote party's commitment. It's kind of confusing to explain succinctly in the comment but I use "other party" to refer to the remote node if `isOurs` is true and the local node if `isOurs` is false.

Reply:
Yeah, that works too. I see `commitmentKeyRing` as a pretty much immutable data struct, so since there are no other methods, I think it would be a bit weird to add one for `paymentKey()`, especially since the overhead of storing the extra pointer is minimal.